### PR TITLE
Relations in status ouput: filter futher.

### PR DESCRIPTION
**Pull request by anastasiamac**
## Description of change

When an operator specifies an application name as a filter to ```juju status```, i.e. ```juju status $APP-NAME```, there could be some applications returned that do not match the filter exactly. They are deemed to be relevant in some for to the requested $APP-NAME. For example, a unit of this application may be deployed to a unit of application $APP-NAME. However, the relations for these 'discovered' applications should not be returned.

In other words, if there are two applications, A and B, such that:
 * an application A matches the supplied filter directly;
 * an application B has units on the same machine as units of an application A and, thus, qualifies to be returned by the status result;
then the application B's relations should not be returned.

## QA steps

Above scenario works.

## Bug reference
n/a

### CODE REVIEW

**Comment by anastasiamac** [comment]
[limit relations in status further: only show relations for application that matches criteria](https://trello.com/c/X7Wz2CPE/246-limit-relations-in-status-further-only-show-relations-for-application-that-matches-criteria)

**Comment by anastasiamac** [comment]
$$merge$$
